### PR TITLE
NNS topology

### PR DESCRIPTION
**Pull request by avivtur**
Adding a first skeleton to the NNS topology, focusing on:
1. NNS CR display and it's interfaces
2. Display green/red border to each interface to represent if the interface is up or down. (if default color appears it means the interface state is unknown)
3. Opening a details drawer on interface select + fill in details.
4. Add topology toolbar to toggle between list and graph views
5. Add filter by node name to see just some of the node.
6. Adding connecting edges between interfaces.

Enhancements to do/Issues to fix:
1. When filtering and selecting few nodes, it always put the node in the center which can cause overlapping nodes which is confusing to look at.
2. Fix single node drag.
3. Improve interfaces drawer info.

Demo:

https://github.com/user-attachments/assets/64c52987-93b8-4628-8aaa-8d6cae16bbb1

### CODE REVIEW

File: src/utils/components/DetailItem/DetailItem.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+          )}
+          {isEdit && showEditOnTitle && (
+            <FlexItem>
+              <Button
+                data-test-id={`${testId}-edit`}
+                isDisabled={isDisabled}
+                isInline
+                onClick={onEditClick}
+                type="button"
+                variant="link"
```
Comment (metalice) [inline src/utils/components/DetailItem/DetailItem.tsx]: ButtonVariant.link

File: src/utils/components/DetailItem/DetailItem.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  const description = (
+    <EditButtonWithTooltip
+      isEditable={!isDisabled}
+      onEditClick={onEditClick}
+      testId={testId}
+      tooltipContent={messageOnDisabled}
+    >
+      {descriptionData ?? NotAvailable}
+    </EditButtonWithTooltip>
+  );
```
Comment (metalice) [inline src/utils/components/DetailItem/DetailItem.tsx]: nested?

File: src/utils/components/DetailItem/DetailItem.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+  return (
+    <DescriptionListGroup className={`pf-c-description-list__group ${className && className}`}>
```
Comment (metalice) [inline src/utils/components/DetailItem/DetailItem.tsx]: use className package , its nicer

File: src/utils/components/DetailItem/DetailItem.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            </FlexItem>
+          )}
+          {isEdit && showEditOnTitle && (
+            <FlexItem>
+              <Button
+                data-test-id={`${testId}-edit`}
+                isDisabled={isDisabled}
+                isInline
+                onClick={onEditClick}
+                type="button"
```
Comment (metalice) [inline src/utils/components/DetailItem/DetailItem.tsx]: i think its redundant

File: src/utils/components/DetailItem/DetailItem.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            </FlexItem>
+          )}
+        </Flex>
+      </DescriptionListTermHelpText>
+
+      <DescriptionListDescription
+        className="pf-c-description-list__description"
+        data-test-id={testId}
+      >
+        {isEdit && !showEditOnTitle ? description : descriptionData}
```
Comment (metalice) [inline src/utils/components/DetailItem/DetailItem.tsx]: description and description data.. please clarify a bit the naming

File: src/utils/components/DetailItem/DetailItemHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  bodyContent: ReactNode;
+  breadcrumb?: string;
+  descriptionHeader: ReactNode;
+  isPopover: boolean;
+  label?: ReactNode;
+  maxWidth?: string;
+  moreInfoURL?: string;
+};
+
+export const DetailItemHeader: FC<DetailItemHeaderProps> = ({
```
Comment (metalice) [inline src/utils/components/DetailItem/DetailItemHeader.tsx:24]: why making such a genric component? cant split for 2 ?

File: src/utils/components/DetailItem/EditButtonWithTooltip.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  const EditButton = () => (
+    <Button
+      onClick={(e: SyntheticEvent<HTMLButtonElement>) => {
+        e.stopPropagation();
+        onEditClick?.();
+      }}
+      data-test-id={testId}
+      isDisabled={!isEditable}
+      isInline
+      variant="link"
```
Comment (metalice) [inline src/utils/components/DetailItem/EditButtonWithTooltip.tsx]: button variant

File: src/utils/components/DetailItem/EditButtonWithTooltip.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    return (
+      <Tooltip content={tooltipContent}>
+        <span>
+          <EditButton />
+        </span>
+      </Tooltip>
+    );
+  }
+
+  return <EditButton />;
```
Comment (metalice) [inline src/utils/components/DetailItem/EditButtonWithTooltip.tsx]: nested? again? lol

File: src/utils/components/DetailItem/DetailItem.tsx (hunk 2 above)
Comment (avivtur) [reply src/utils/components/DetailItem/DetailItem.tsx]: the DetailItem components was copied from kubevirt-plugin (VirtualMachineDescriptionItem)

File: src/utils/components/DetailItem/DetailItemHeader.tsx (hunk 6 above)
Comment (avivtur) [reply src/utils/components/DetailItem/DetailItemHeader.tsx:24]: also copied from kubevirt

File: src/utils/components/DetailItem/EditButtonWithTooltip.tsx (hunk 8 above)
Comment (avivtur) [reply src/utils/components/DetailItem/EditButtonWithTooltip.tsx]: also copied from kubevirt, again lol

**Review by upalatucci — APPROVED** [review]

**Comment by upalatucci** [comment]
Reviewed. For me it's great thanks aviv

**Comment by avivtur** [comment]
@pcbailey @metalice 
PR was merged by accident, please review, leave comments here and will do a follow-up PR

**Pull request merged**